src/main/java/org/broadinstitute/dsm/jobs/TestBostonUPSTrackingJob.java

```diff
@@ -205,77 +214,121 @@ private static void updateStatus(@NonNull Connection conn, String trackingId, St
-                        EventUtil.triggerDDP(conn, kitDDPNotification);
-                    }
-                    else {
-                        logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
-                    }
+    private void updateTrackingInfo(PoolingDataSource<PoolableConnection> dataSource,
+                                    String upsHistory,
+                                    String statusType,
+                                    UPSActivity lastActivity,
+                                    String trackingId,
+                                    String query,
+                                    boolean isReturn,
+                                    DdpKit kit,
+                                    Instant earliestInTransitTime,
+                                    String ddpInstanceId,
+                                    Config cfg) {
+        final String SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER = "select  eve.*,   request.ddp_participant_id,   request.ddp_label,   request.dsm_kit_request_id, request.ddp_kit_request_id, request.upload_reason, " +
+                "        realm.ddp_instance_id, realm.instance_name, realm.base_url, realm.auth0_token, realm.notification_recipients, realm.migrated_ddp, kit.receive_date, kit.scan_date" +
+                "        FROM " + STUDY_MANAGER_SCHEMA + "ddp_kit_request request, " + STUDY_MANAGER_SCHEMA + "ddp_kit kit, " + STUDY_MANAGER_SCHEMA + "event_type eve, " + STUDY_MANAGER_SCHEMA + "ddp_instance realm where request.dsm_kit_request_id = kit.dsm_kit_request_id and request.ddp_instance_id = realm.ddp_instance_id" +
+                "        and not exists " +
+                "                    (select 1 FROM " + STUDY_MANAGER_SCHEMA + "EVENT_QUEUE q" +
+                "                    where q.DDP_INSTANCE_ID = realm.ddp_instance_id" +
+                "                    and " +
+                "                    q.EVENT_TYPE = eve.event_name" +
+                "                    and " +
+                "                    q.DSM_KIT_REQUEST_ID = request.dsm_kit_request_id " +
+                "                    and q.event_triggered = true" +
+                "                    )" +
+                "        and (eve.ddp_instance_id = request.ddp_instance_id and eve.kit_type_id = request.kit_type_id) and eve.event_type = ? " +
+                "         and realm.ddp_instance_id = ?" +
+                "          and kit.dsm_kit_request_id = ?";
+        Connection conn = null;
+        try {
+            conn = dataSource.getConnection();
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                stmt.setString(1, kit.getDsmKitRequestId());
+                stmt.setString(2, upsHistory);
+                stmt.setString(3, upsHistory);
+                int r = stmt.executeUpdate();
+                conn.commit();
+                logger.info("Updated " + r + " rows");
```

Comment:
I'm not sure the reader of the logs will know what's going on without more context-- `Updated 3 rows` is somewhat ambiguous...